Ldelossa/phase out sec label pt1

ldelossa opened the pull request:
Currently in the datapath we swap around the terminology `sec_label` and `identity`. 

As an effort to consolidate our terminology, this PR begins the phasing out of the keywords `sec_label` in favor of `sec_identity`.

An arbitrary starting point was picked, the `remote_endpoint_info` struct, and further renames will take place over time, as to not create too many conflicts with other PRs. 

```release-note
Rename the `sec_label` field in remote_endpoint_info structure to `sec_identity`
```

aspsk commented on bpf/bpf_host.c:
```diff
@@ -101,18 +101,18 @@ ipcache_lookup_srcid6(struct __ctx_buff *ctx)
 	struct remote_endpoint_info *info = NULL;
```
`also update all update variables` -- too many `update`s

ldelossa replied:
Fixed in newest commit

aspsk commented on bpf/lib/host_firewall.h:
```diff
@@ -16,417 +16,417 @@
 
 # ifdef ENABLE_IPV6
 static __always_inline int
-ipv6_host_policy_egress(struct __ctx_buff *ctx, __u32 src_id,
```
This diff is huge because tabs were replaced by spaces (in other places too). Please, push a new version

ldelossa replied:
Ugh yeah, wrong editor profile. Will fix.

ldelossa committed:
bpf: rename remote_endpoint_info.sec_label to sec_identity

 In the datapath we often swap between calling the `security identity` an
`identity` and a `sec_label`.

As an effort to consolidate our terminology, begin phasing out the
`sec_label` terminology, with the first step focusing on the
`remote_endpoint_info` structure.

Rename the `sec_label` field on this structure to `sec_identity`, and also
update variables to consistently use `*_sec_identity` syntax.

Signed-off-by: Louis DeLosSantos <louis.delos@isovalent.com>

aspsk reviewed (approved):
LGTM from technical perspective.

I am not a big fan of this `_sec_` thing though. Do we have other types of identities? If we want to emphasize that this is a "Secutiry Identity" then I would rather use `security_id` than `sec_identity`.
